clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -475,11 +500,18 @@ public Node leastLoadedNode(long now) {
     }
 
     public static AbstractResponse parseResponse(ByteBuffer responseBuffer, RequestHeader requestHeader) {
+        return parseResponseMaybeUpdateThrottleTimeMetrics(responseBuffer, requestHeader, null, 0);
```

Comment:
@junrao At authentication time, we don't know which throttle time sensor will be used (fetch/produce). `SaslClientAuthenticator` sets `throttleTimeSensor` to null and doesn't record throttle time. In any case, we don't currently throttle requests prior to authentication, but the authentication request time is included towards calculation in the broker. Subsequent requests may be throttled as a result and those times are included in the client metrics using the appropriate sensor. Does that make sense? Thanks.